src/core/unit.c

```diff
@@ -1590,6 +1617,12 @@ int unit_start(Unit *u) {
         if (!unit_supported(u))
                 return -EOPNOTSUPP;
 
+        /* Let's make sure that the deps really are in order before we start this. Normally the job engine should have
+         * taken care of this already, but let's check this here again. After all, the dependencies might have changed
+         * due to a reload or because the user forgot to place ordering dependencies in conjunction with BindsTo=. */
```

Comment:
Hm, isn't this comment out of date now? We don't require ordering dependencies, so saying that the user "forgot" is misleading. Also, if there are no ordering dependencies, this does not change the requirement dependencies, so that last sentence is wrong ("the dependencies might have changed ... because the user forgot to place ordering dependencies").